cartography/intel/gcp/crm.py

```diff
@@ -135,42 +136,62 @@ def load_gcp_projects(neo4j_session, data, gcp_update_tag):
     :param gcp_update_tag: The timestamp value to set our new Neo4j nodes with
     :return: Nothing
     """
+    query = """
+    MERGE (project:GCPProject{id:{ProjectId}})
+    ON CREATE SET project.firstseen = timestamp()
+    SET project.projectid = {ProjectId},
+    project.projectnumber = {ProjectNumber},
+    project.displayname = {DisplayName},
+    project.lifecyclestate = {LifecycleState},
+    project.lastupdated = {gcp_update_tag}
+    """
+
     for project in data:
-        if project.get('parent', None):
-            if project['parent']['type'] == "organization":
-                query = """
-                MERGE (parent:GCPOrganization{id:{ParentId}})
-                ON CREATE SET parent.firstseen = timestamp()
-                """
-                parentid = f"organizations/{project['parent']['id']}"
-            elif project['parent']['type'] == "folder":
-                query = """
-                MERGE (parent:GCPFolder{id:{ParentId}})
-                ON CREATE SET parent.firstseen = timestamp()
-                """
-                parentid = f"folders/{project['parent']['id']}"
-        query += """
-        MERGE (project:GCPProject{id:{ProjectId}})
-        ON CREATE SET project.firstseen = timestamp()
-        SET project.projectid = {ProjectId},
-        project.projectnumber = {ProjectNumber},
-        project.displayname = {DisplayName},
-        project.lifecyclestate = {LifecycleState},
-        project.lastupdated = {gcp_update_tag}
-        WITH parent, project
-        MERGE (parent)-[r:RESOURCE]->(project)
-        ON CREATE SET r.firstseen = timestamp()
-        SET r.lastupdated = {gcp_update_tag}
-        """
         neo4j_session.run(
             query,
-            ParentId=parentid,
             ProjectId=project['projectId'],
             ProjectNumber=project['projectNumber'],
             DisplayName=project.get('name', None),
             LifecycleState=project.get('lifecycleState', None),
             gcp_update_tag=gcp_update_tag,
         )
+        if project.get('parent'):
+            _attach_gcp_project_parent(neo4j_session, project, gcp_update_tag)
```

Comment:
@ramonpetgrave64 I did the attach thing, whatchu think?

Reply:
Looks good, and it conforms better with the other` _attach` functions.